### PR TITLE
feat(tools): check mod exports 

### PR DESCRIPTION
**Pull request by timreichen**
checks if `mod.ts` files export all `.ts` files excluding prefixed `.*` `_*` files as well as `test.ts` and `*_test.ts` files.
This should prevent problems when new files are added to std.

**Note** There are some missing exports atm. The failed lint test demonstrates what this tool looks like in action. The missing files should be added regardless, but could also be added with this PR.

### CODE REVIEW

**Comment by kt3k** [comment]
I think `front_matter/{yaml,json,toml,any}.ts` should be added to the ignore list. The exports from these files conflict with existing exported items.

Also the entire `internal` dir should be added to the ignore list as they are not part of public API.

I'm not sure about `yaml/schema.ts`, but it feels like not a public API. Probably we should prefix this file with `_`.

> Note There are some missing exports atm. The failed lint test demonstrates what this tool looks like in action. The missing files should be added regardless, but could also be added with this PR.

These need to be addressed before this PR merged. We don't merge commit which makes main CI red.

**Comment by timreichen** [comment]
> Also the entire `internal` dir should be added to the ignore list as they are not part of public API.

Why is `internal` public at all and not underscored? Seems like a naming contradiction. Also the `mod.ts` says, it is only for internal use.

> I'm not sure about `yaml/schema.ts`, but it feels like not a public API. Probably we should prefix this file with `_`.

Agreed.

> > Note There are some missing exports atm. The failed lint test demonstrates what this tool looks like in action. The missing files should be added regardless, but could also be added with this PR.
> 
> These need to be addressed before this PR merged. We don't merge commit which makes main CI red.

Sorry, what I meant was: Should we do it in another PR or in this?

> I think `front_matter/{yaml,json,toml,any}.ts` should be added to the ignore list. The exports from these files conflict with existing exported items.

I wonder what the conditions are whether a mod has a `mod.ts` or just single file exports. I don't think it makes much sense to have a `mod.ts` if it doesn't export the (mostly) all of the apis or at least not the "main" apis.

### Big sidenote
There is a general design problem how deno handles test files (e.g. `test.ts`,  `*_test.ts`) because snake case is how all files should be named according to the [style guide](https://docs.deno.com/runtime/manual/references/contributing/style_guide#use-underscores-not-dashes-in-filenames), yet apis with `test` in the name are prone to name conflicts where they cannot be differentiated, so either they falsely run with `deno test` or are ignored by tools like these. That happens now with front_matter/test.ts which is not a test file, though deno sees it as one. Is the core team aware of that and is this a concern?

**Comment by kt3k** [comment]
> Sorry, what I meant was: Should we do it in another PR or in this?

Let's do them in this PR

**Comment by kt3k** [comment]
`front_matter/{yaml,json,toml,any}.ts` These 4 files all export the same name API `extract`. I'm not sure how to re-export them from `mod.ts` (I guess re-exporting from mod.ts doesn't make much sense for them)

**Comment by timreichen** [comment]
> `front_matter/{yaml,json,toml,any}.ts` These 4 files all export the same name API `extract`. I'm not sure how to re-export them from `mod.ts` (I guess re-exporting from mod.ts doesn't make much sense for them)

Ok, I guess we could rename exports like `extrqctFromYaml()`,but i am also fine with excluding them.

What about `internal`?

**Comment by kt3k** [comment]
> Ok, I guess we could rename exports like extractFromYaml(),but i am also fine with excluding them.

Let's exclude them for now.

> What about internal?

We are going to switch the registry from `deno.land/std` to a new one at Deno 2.0 (std 1.0), and the new registry currently doesn't allow submodule starting with `_` (This could be changed later). So let's keep it excluded for now.

File: yaml/mod.ts
Comment (timreichen) [reply yaml/mod.ts:54]: wouldn't that be a breaking change?

Comment (kt3k) [reply yaml/mod.ts:54]: Ok. Let's address that in another issue. For now please add `schema.ts` to the exclude list instead of re-exporting here as we don't want to encourage the usage of items in it.

**Review by kt3k — APPROVED** [review]
LGTM. Thanks! Looks like a nice check to have

**Pull request merged**
